### PR TITLE
load markers with ajax

### PR DESCRIPTION
**Pull request by PeterTheOne**
This PR adds a map that gets its coordinates via ajax request, depending on the current bounding box of the map.

In more detail: It adds a new map ParserHook (`display_ajax_map`) that passes the location unprocessed to the mapdata json div. Javascript listens to `dragend` and `zoom_changed` events of the map and fetches new coordinates with an ajax request to the Ask API. Then the marker points are added to the coordinates.

```
{{#display_ajax_map:
 [[Category:Locations]]
 |service=googlemaps
}}
```

`query:[[Category:Locations]] [[Has coordinates::+]] [[Has coordinates::>45.31645348670738°, -1.6738700866699219°]] [[Has coordinates::<50.46714516668226°, 29.966754913330078°]]|?Has coordinates`

This PR tries to solve the problem of having a lot of locations that result in a lot of markers on the map at the same time, which causes performance problem. By only loading locations/markers that are in the field of view of the map (bounding box) the marker count is reduced.

I hope this makes sense. For now it only works for the googlemaps service and it still has some rough edges so feedback is very welcome! I'm also willing to squash my commits if needed.

Depends on https://github.com/JeroenDeDauw/Maps/pull/117 .

### CODE REVIEW

File: src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ *
+ * @licence GNU GPL v2+
+ * @author Peter Grassberger <petertheone@gmail.com>
+ */
+
+function ajaxUpdateMarker(googlemaps) {
+    var bounds = googlemaps.map.getBounds();
+
+    var query = googlemaps.options.coordinates.join(' ') + ' ';
+    query += '[[Has coordinates::+]] ';
```
Comment (mwjames) [inline src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js]: `[[Has coordinates::+]]`, I do not think this is a viable approach as properties are hard coded which will fail for environments that don't declare or use different properties for the representation of coordinates.

File: src/parserhooks/SM_DisplayAjaxMap.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ * Class for the 'display_map_ajax' parser hooks.
+ *
+ * @since 0.7
+ *
+ * @licence GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ * @author Peter Grassberger < petertheone@gmail.com >
+ */
+class SMDisplayAjaxMap extends MapsDisplayMap {
+
```
Comment (mwjames) [inline src/parserhooks/SM_DisplayAjaxMap.php]: I'm not sure that this class should reside within SemanticMaps because it doesn't rely on any "semantic" related functions and I would guess  it more appropriate for Maps.

@JeroenDeDauw ?

File: src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js (hunk 1 above)
Comment (JeroenDeDauw) [inline src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js]: Yeah, this is correct

File: src/parserhooks/SM_DisplayAjaxMap.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            'islist' => true,
+            'delimiter' => $type === ParserHook::TYPE_FUNCTION ? ';' : "\n",
+            'message' => 'maps-displaymap-par-coordinates',
+        );
+
+        $params['ajax'] = array(
+            'type' => 'boolean',
+            'default' => true,
+            'message' => 'maps-displaymap-par-ajax',
+        );
```
Comment (JeroenDeDauw) [inline src/parserhooks/SM_DisplayAjaxMap.php]: Why did you create a new parser hook, rather than just adding a flag like this one to the existing one? This flag seems to be currently unused.

File: src/parserhooks/SM_DisplayAjaxMap.php (hunk 2 above)
Comment (JeroenDeDauw) [inline src/parserhooks/SM_DisplayAjaxMap.php]: If it is only used by SM, then I don't see a problem with it being there. I'd only put it in Maps if there where some kinds of plans to use it there. If it is in SM, it can be moved down if needed anyway, sparing some extra maintenance burden caused by separated code till then.

File: src/parserhooks/SM_DisplayAjaxMap.php (hunk 3 above)
Comment (JeroenDeDauw) [inline src/parserhooks/SM_DisplayAjaxMap.php]: I just realized that there is no "show map" parser function in SM. You'd have to use the map SMW result format. Is there a reason to not use that one?

File: src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js (hunk 1 above)
Comment (JeroenDeDauw) [inline src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js]: This begs the question of how to design the parser hook / result format: how and what information will the user provide?

It seems like you need a base query (in the hardcoded case here being `[[Has coordinates::+]]`), and the name of the coordinate property to build the bounding box with and select. Perhaps something like this?

```
| morelocationsquery = [[:Category:City]] [[Has coordinates::+]]
| boundingboxproperty = Has coordinates
```

(Note that these are some random names I thought up and that they don't look nice at all.)

Perhaps the base query can just be what was already provided to the ask parser function? I don't remember if the result format has access to that.

As the maps result format can deal with more than just locations (additional info displayed as tooltips), it is good to retain this behaviour when fetching additional data. Again access to the initial query would be nice, so the properties one wants to select do not need to be repeated.

Ping @mwjames

File: src/parserhooks/SM_DisplayAjaxMap.php (hunk 3 above)
Comment (PeterTheOne) [inline src/parserhooks/SM_DisplayAjaxMap.php]: > rather than just adding a flag like this one to the existing one

The first thing I tried was adding the ajax property to `Maps_DisplayMapRenderer`. Now the property is unused and should be removed.

> I just realized that there is no "show map" parser function in SM. You'd have to use the map SMW result format. Is there a reason to not use that one?

I haven't looked into that yet.

File: src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js (hunk 1 above)
Comment (PeterTheOne) [inline src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js]: Thanks for your Feedback :) Let me rephrase to make sure I understand correctly.

The coordinates property is hardcoded now, I would like to make the user specify it with: `| coordinatesproperty = Has coordinates` (you named it `boundingboxproperty`). And then use it for the boundingbox (`[[Has coordinates::<...]]`), to check if it exists (`[[Has coordinates::+]]`) and to return it (`|?Has coordinates`) because these are to base requirements (basequery?) that are needed fetch coordinates and then add to marker.

I'm not sure what you mean with `morelocationsquery` because the user can specify the query just like this:

```
{{#display_ajax_map:
 [[Category:City]]
 | coordinatesproperty = Has coordinates
 | service = googlemaps
}}
```

> Perhaps the base query can just be what was already provided to the ask parser function? I don't remember if the result format has access to that.

I'm not sure what you have in mind here.

> As the maps result format can deal with more than just locations (additional info displayed as tooltips), it is good to retain this behaviour when fetching additional data. Again access to the initial query would be nice, so the properties one wants to select do not need to be repeated.

Yes, it would be good to retain this behavior, I will look into this. But I don't get what you mean by initial query.

File: src/parserhooks/SM_DisplayAjaxMap.php (hunk 2 above)
Comment (PeterTheOne) [inline src/parserhooks/SM_DisplayAjaxMap.php]: I thought about adding this functionality to Maps also, but for now I think it makes more sense for SemanticMaps only. Reasoning behind this was: For Semantic Maps I'm using the Ask API to fetch data and without this API I would need to write something like it for non semantic data that lies in one page.

**Comment by JeroenDeDauw** [comment]
Sorry for not getting back to you yet, I'm leaving this tab open and will hopefully get to it tomorrow.

File: src/parserhooks/SM_DisplayAjaxMap.php (hunk 3 above)
Comment (PeterTheOne) [inline src/parserhooks/SM_DisplayAjaxMap.php]: Since 52098cd2a698fb8925e40ace3d3c91891209ddee the ajax property  is in use again.

File: src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js (hunk 1 above)
Comment (PeterTheOne) [inline src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js]: In f6b9bf150837c2c07de6b718aa0b28dae984d093 I added the `coordinatesproperty` property.

**Comment by mwjames** [comment]
I don't want to crash the party here but you might want to check out https://github.com/SemanticMediaWiki/WhatsNearby, I think it  brings an ajax maps feature and includes some other neat options such as changing limit or radius dynamically, use of localcache etc.. 

```
{{#nearby: [[Has coordinates::50° 50' 48" N, 4° 21' 10" E  (4200 km)]]
 |?Has coordinates
 |limit=5
 |max=100
 |interval=10
 |nolocation=true
 |format=openlayers
 |querytemplate=nearby-standard-ask-query
 |localcache=no
 |controls=slider
}}

{{#nearby: [[Has text::~Lorem ipsum]]
 |?Has text
 |limit=5
 |max=100
 |interval=10
 |nolocation=true
 |format=table
 |localcache=100
 |controls=slider
 |querytemplate=nearby-standard-ask-query
}}
```

image

@kghbln Maybe we can add `mediawiki/whats-nearby:@dev` to the sandbox which requires Maps@dev and SMW 2.4 (as soft dependency).

**Comment by kghbln** [comment]
> Maybe we can add mediawiki/whats-nearby:@dev to the sandbox which requires Maps@dev and SMW 2.4 (as soft dependency). 

Note: All extensions installed by composer are at @dev . No idea why it does not show properly on Special:Version or alternatively why composer does not honour the json-file. Howver, another story.

**Comment by PeterTheOne** [comment]
WhatsNearby looks interesting, I will try it.

**Comment by krabina** [comment]
looks very cool, but when I try to install it, I get: 

- The requested package mediawiki/whats-nearby could not be found in any version, there may be a typo in the package name. 

cheers, 
Bernhard 

----- Am 31. Mrz 2016 um 19:27 schrieb mwjames <notifications@github.com>:

**Comment by mwjames** [comment]
>  No idea why it does not show properly on Special:Version

That's an MW issue caused by the cache, It should be visible [now](http://sandbox.semantic-mediawiki.org/wiki/Sp%C3%A9cial:Version).

However, the sandbox Maps is still on 3.4.1 but `whats-nearby`requires https://github.com/JeroenDeDauw/Maps/pull/105 (which is in @dev).

> The requested package mediawiki/whats-nearby could not be found in any version,

You need the `mediawiki/whats-nearby:@dev` (dev) version as it is not yet released [0].

[0] https://packagist.org/packages/mediawiki/whats-nearby

**Comment by mwjames** [comment]
> looks very cool, but when I try to install it, I get:
The requested package mediawiki/whats-nearby could not be found in any version, there may be a typo in the package name.

For people interested in sneak-preview, [0] has some examples.

[0] http://sandbox.semantic-mediawiki.org/wiki/Cat%C3%A9gorie:Whats_Nearby_example

**Comment by krabina** [comment]
have it installed now. 

But with the current doc, I am not able to get it up and running. Is there a public wiki where usage is shown? 

----- Am 1. Apr 2016 um 14:10 schrieb mwjames <notifications@github.com>:

**Comment by mwjames** [comment]
> But with the current doc, I am not able to get it up and running.

There is always room for improvement and help is welcome to make it more user friendly. 

> Is there a public wiki where usage is shown?

Did you see the link to [0]?

[0] http://sandbox.semantic-mediawiki.org/wiki/Cat%C3%A9gorie:Whats_Nearby_example

File: src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js (hunk 1 above)
Comment (JeroenDeDauw) [inline src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js]: What I was talking about does not fully make sense if you actually add a new parser function as you are doing with `display_ajax_map`. It's still unclear to me why you are going with that approach, rather than extending the existing map query result format.

```
{{#ask: [[:Category:City]] [[Has coordinates::+]]
| morelocationsquery = [[:Category:City]] [[Has coordinates::+]]
| boundingboxproperty = Has coordinates
| ?Has coordinates
| ?Has description=Description
| ?Has old coordinates=Previous location  
}}
```

**Comment by krabina** [comment]
thank you for that, this helped. 

It's really awesome, thank you James!!!! 

Here is my little test: http://standards.kdz.eu/index.php?title=LocationTest 

cheers, 
Bernhard 

----- Am 1. Apr 2016 um 16:20 schrieb mwjames <notifications@github.com>:

**Comment by PeterTheOne** [comment]
Usage:

    {{
    #ask:
    [[Category:Locations]] [[Has coordinates::47° 4' 0", 15° 26' 0" E  (50 km)]]
    | ?Has coordinates
    | format=map
    | service=googlemaps
    | ajaxquery = [[Category:Locations]]
    | coordinatesproperty = Has coordinates
    }}

The first query `[[Category:Locations]] [[Has coordinates::47° 4' 0", 15° 26' 0" E  (50 km)]]` is used on site load. The last part of the query limits the distance of points to be loaded.

On zoom or drag of the map the second query `[[Category:Locations]]` is used to load points in the current position (boundingbox) of the map.

**Comment by JeroenDeDauw** [comment]
Looks mostly good

File: src/queryprinters/SM_MapPrinter.php [hunk 4]
```diff
@@ -96,6 +96,16 @@ protected function getParameterInfo() {
 			'default' => false,
 		];
 
+		$params['coordinatesproperty'] = array(
```
Comment (JeroenDeDauw) [inline src/queryprinters/SM_MapPrinter.php]: What about calling this ajaxproperty? That is presumably more clear to the user, and less prone to lead to confusion.

@kghbln thoughts?

File: src/queryprinters/SM_MapPrinter.php [hunk 5]
```diff
@@ -123,6 +133,7 @@ protected function getParameterInfo() {
 		if ( $this->fatalErrorMsg !== false ) {
 			return $this->fatalErrorMsg;
 		}
+		$this->isHTML = true;
```
Comment (JeroenDeDauw) [inline src/queryprinters/SM_MapPrinter.php]: Why is this change needed?

File: src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+function ajaxUpdateMarker(map) {
+    var bounds = map.map.getBounds();
+
+    var coordinatesproperty = map.options.coordinatesproperty;
+
+    var query = map.options.ajaxquery.join(' ') + ' ';
+    query += '[[' + coordinatesproperty + '::+]] ';
+    query += '[[' + coordinatesproperty + '::>' + bounds.getSouthWest().lat() + '°, ' + bounds.getSouthWest().lng() + '°]] ';
+    query += '[[' + coordinatesproperty + '::<' + bounds.getNorthEast().lat() + '°, ' + bounds.getNorthEast().lng() + '°]]';
+    query += '|?' + coordinatesproperty;
```
Comment (JeroenDeDauw) [inline src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js]: This can be extracted into a `getQueryString` or some such function

File: src/queryprinters/SM_MapPrinter.php (hunk 4 above)
Comment (PeterTheOne) [inline src/queryprinters/SM_MapPrinter.php]: I'm open to renaming it and it would be good to hint at it being used for the ajax query but it also should be clear that it is the coordinates property.

File: src/queryprinters/SM_MapPrinter.php (hunk 5 above)
Comment (PeterTheOne) [inline src/queryprinters/SM_MapPrinter.php]: This is needed because the value of `ajaxquery` would be parsed and turn out as an empty string.

File: src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js (hunk 6 above)
Comment (PeterTheOne) [inline src/services/GoogleMaps3/ext.sm.googlemaps3ajax.js]: ok sure.

File: src/queryprinters/SM_MapPrinter.php (hunk 5 above)
Comment (JeroenDeDauw) [inline src/queryprinters/SM_MapPrinter.php]: This also makes the other things not being parsed then, breaking existing functionality, no? IIRC there is some way to not have something be parsed, by putting special tags around it.

Comment (PeterTheOne) [inline src/queryprinters/SM_MapPrinter.php]: > breaking existing functionality, no?

Haven't noticed. If there is a way without `$this->isHTML = true;` please tell me.

Comment (JeroenDeDauw) [inline src/queryprinters/SM_MapPrinter.php]: Can you try using this together with the `intro` parameter that contains some wikitext and see if it gets rendered properly?

Comment (mwjames) [inline src/queryprinters/SM_MapPrinter.php]: Yes, I would be very cautions about this change which might have unanticipated consequences for when templates are rendered.

Comment (PeterTheOne) [inline src/queryprinters/SM_MapPrinter.php]: It doesn't. :(

If I remove `$this->isHTML = true;` I can use `<nowiki>` to make it work, but it's not convenient.

    {{
    #ask:
    [[Category:Locations]] [[Has coordinates::47° 4' 0", 15° 26' 0" E  (50 km)]]
    | ?Has coordinates
    | format=googlemaps
    | ajaxquery = <nowiki>[[Category:Locations]]</nowiki>
    | coordinatesproperty = Has coordinates
    | intro = some [[Graz]] thing:
    }}

Comment (JeroenDeDauw) [inline src/queryprinters/SM_MapPrinter.php]: How about adding the `nowiki` tags in the code, right before where they are needed? Forcing the user to put it in the wikitext is indeed terrible.

Comment (PeterTheOne) [inline src/queryprinters/SM_MapPrinter.php]: > How about adding the nowiki tags in the code, right before where they are needed?

Is this even possible? It's gone before I can touch it as far as I tried..

File: src/services/Leaflet/ext.sm.leafletajax.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * @licence GNU GPL v2+
+ * @author Peter Grassberger < petertheone@gmail.com >
+ */
+
+
+(function( $ ) {
+    var ajaxRequest = null;
+
+    // todo: maybe move to Maps
```
Comment (JeroenDeDauw) [inline src/services/Leaflet/ext.sm.leafletajax.js]: Maps is not supposed to know about SMW, so this should not be moved there

File: src/queryprinters/SM_MapPrinter.php (hunk 5 above)
Comment (JeroenDeDauw) [inline src/queryprinters/SM_MapPrinter.php]: I quickly looked at the code. Perhaps changing `getJSONObject` to this works? I'm really not up to speed with the code so am not sure this is the correct place (as I don't know where the parse you are tripping over is happening).

```php
protected function getJSONObject( array $params, Parser $parser ) {
		if ( array_key_exists( 'ajaxquery', $params ) ) {
			$params['ajaxquery'] = '<nowiki>' . $params['ajaxquery'] . '</nowiki>';
		}

		return $params;
	}
```

Comment (JeroenDeDauw) [inline src/queryprinters/SM_MapPrinter.php]: Having this `'ajaxquery'` string at multiple places is not really nice btw. You could have a constant in the class instead. I know that is not what the other code is doing either, though it looks like it should. Just a suggestion

Comment (PeterTheOne) [inline src/queryprinters/SM_MapPrinter.php]: The param `$params['ajaxquery']` is empty before that, I end up with `<nowiki></nowiki>`. :(

Comment (PeterTheOne) [inline src/queryprinters/SM_MapPrinter.php]: Still haven't found a solution for this..

Comment (JeroenDeDauw) [inline src/queryprinters/SM_MapPrinter.php]: Do you know what code makes it become empty? The `nowiki` tags need to be added right before.

**Comment by PeterTheOne** [comment]
Ajax loading of markers now works for all services (googlemaps, leaflet and openlayers)! :)

**Comment by JeroenDeDauw** [comment]
That's great! Still blocked on https://github.com/JeroenDeDauw/Maps/pull/161 and the HTML/parse issue.

File: src/queryprinters/SM_MapPrinter.php (hunk 5 above)
Comment (PeterTheOne) [inline src/queryprinters/SM_MapPrinter.php]: No clue, I guess it is somewhere deep down in the Parser functions? Or is it possible to set the param type to something else that won't be parsed? Or could I create a ParserHook that would do the trick?

Comment (Flexmaen) [inline src/queryprinters/SM_MapPrinter.php]: Why not just having a syntax like
_| ajaxquery = on_

If you have it like
_| ajaxquery = [[Category:Locations]]_
this means you have to specify the same query for a 2nd time. Well, you could also have a different query here than you have on the top of the ask block. But I fail to see the usecase where this makes sense. Is there a specific reason, why it just can't be done like "ajaxquery = on"?

**Comment by PeterTheOne** [comment]
https://github.com/JeroenDeDauw/Maps/pull/161 is merged, only the HTML/parse issue left.

File: src/queryprinters/SM_MapPrinter.php (hunk 5 above)
Comment (kghbln) [inline src/queryprinters/SM_MapPrinter.php]: > `| ajaxquery = true`

"yes/no" would be the pair for reasons of consistency to other query parameters. This should default to "no"

> this means you have to specify the same query for a 2nd time.

This indeed does not seem to be necessary to me too. This may very well be considered if it helps getting out of this situation. @PeterTheOne There was probably a reason for this in the first place I am currently not seeing? @JeroenDeDauw Thoughts on this?

Comment (PeterTheOne) [inline src/queryprinters/SM_MapPrinter.php]: > this means you have to specify the same query for a 2nd time.

I think @JeroenDeDauw suggested it. I'm fine with changing this, especially when it solves the `isHTML`/`<nowiki>` problem! But you would have to specify the range of the first query anyhow `[[Has coordinates::47° 4' 0", 15° 26' 0" E  (50 km)]]` and more importantly I still don't know how to get at the first query's data to pass it to my javascript ajax function. Any help?

Comment (kghbln) [inline src/queryprinters/SM_MapPrinter.php]: Ouch, I now see the misery we are in here. Let's wait for @JeroenDeDauw to be the White Knight. I am a bit "unterbelichtet" here to put it in kind words.

**Comment by PeterTheOne** [comment]
I've removed `isHTML = true` for now, you can only use the ajaxquery like this `| ajaxquery = <nowiki>*query-goes-here*</nowiki>`.

**Comment by JeroenDeDauw** [comment]
If we add the feature like that, then we'll need to keep supporting the case where those tags are included, even when we remove the need for them. Not a deal breaker, though not nice.

Have you tried asking for help on the wikitech-l mailing list, or other MediaWiki dev channels? I might be able to try find the issue in a few days, no promises though.

File: src/queryprinters/SM_MapPrinter.php (hunk 5 above)
Comment (JeroenDeDauw) [inline src/queryprinters/SM_MapPrinter.php]: You might be able to get at the query string in the rest printer. I don't recall. More of a question for @mwjames.

Looking at the code, this looks like that might be it: 

```php
SMWQueryResult $res

$res->getQuery()->getQueryString();
```

By the looks of it, that will just give you the "description" part of the query, which is what you want here. (Sad me is sad about https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/127 not being done)

Comment (mwjames) [inline src/queryprinters/SM_MapPrinter.php]: > $res->getQuery()->getQueryString();

As the name suggests, it gets you the string.

> that will just give you the "description"

`$res->getQuery()->getDescription()`

> Sad me is sad about SemanticMediaWiki/SemanticMediaWiki#127 not being done)

Don't blame me ...

**Comment by PeterTheOne** [comment]
> If we add the feature like that, then we'll need to keep supporting the case where those tags are included, even when we remove the need for them. Not a deal breaker, though not nice.

I was kind of surprised but there were no changes needed to support the `<nowiki>` tag, so there are no changes needed to support this case later on.

> Have you tried asking for help on the wikitech-l mailing list, or other MediaWiki dev channels? I might be able to try find the issue in a few days, no promises though.

I haven't yet, I'm kind of new to the community and don't know about all the channels and lists.. ^^

**Comment by JeroenDeDauw** [comment]
> I was kind of surprised but there were no changes needed to support the <nowiki> tag, so there are no changes needed to support this case later on.

My point is that when we make the text not be parsed properly, then the nowiki tags will also not get parsed and just be embedded, unless we deal with them.

> I haven't yet, I'm kind of new to the community and don't know about all the channels and lists.. ^^

https://www.mediawiki.org/wiki/Mailing_lists

**Comment by PeterTheOne** [comment]
> My point is that when we make the text not be parsed properly, then the nowiki tags will also not get parsed and just be embedded, unless we deal with them.

Ok, you could be right. (With `isHTML = true` and `<nowiki>` it still works.)

**Comment by PeterTheOne** [comment]
As written in https://github.com/SemanticMediaWiki/SemanticMaps/pull/52#discussion_r67227021 I've tried to add `<nowiki>` somewhere in https://github.com/SemanticMediaWiki/SemanticMaps/blob/master/src/queryprinters/SM_MapPrinter.php#L138 but it doesn't have any effect and gets escaped.

**Comment by JeroenDeDauw** [comment]
So it is broken at the moment? This PR is really dragging out a long time

**Comment by PeterTheOne** [comment]
The current state is: Works with manuelly adding `<nowiki>` to ajaxquery.

**Comment by JeroenDeDauw** [comment]
Do you want that merged? Or do you want to remove that parameter and use the description of the original query?

**Comment by PeterTheOne** [comment]
I want it merged like this. And I hope we find a way to remove `<nowiki>` soon..
